### ISSUE TITLE
Is there a bug about cyclegan?

### ISSUE THREAD

**Issue by onepeachbiubiubiu** [GaParmar/img2img-turbo#56]
I noticed that the training code does not set ‘requires_grad=False’ for the discriminator when training generators, which will conduct gradients to discriminators and force the discriminator to treat the fake image as a 'true image'.

**Comment by GaParmar** [comment]
I am a little confused by this. Could you point to the relevant line in the training code? 

Here, the fake image is `detached` which should stop the flow of gradients:
https://github.com/GaParmar/img2img-turbo/blob/main/src/train_cyclegan_turbo.py#L232

-Gaurav

**Comment by onepeachbiubiubiu** [comment]
https://github.com/GaParmar/img2img-turbo/blob/main/src/train_cyclegan_turbo.py#L203

**Comment by onepeachbiubiubiu** [comment]
When training the generator, the discriminator's 'requires_grad' is Ture.  So https://github.com/GaParmar/img2img-turbo/blob/main/src/train_cyclegan_turbo.py#L205 will produce gradients for the discriminator. The gradients will update to the discriminator's parameter in https://github.com/GaParmar/img2img-turbo/blob/main/src/train_cyclegan_turbo.py#L238.

**Comment by yahshibu** [comment]
I think that is the case for train_pix2pix_turbo.py.

Gradients for the discriminator will be produced at
https://github.com/GaParmar/img2img-turbo/blob/main/src/train_pix2pix_turbo.py#L200 .
And, the gradients won't be deleted until
https://github.com/GaParmar/img2img-turbo/blob/main/src/train_pix2pix_turbo.py#L217 .

I think
`accelerator.unwrap_model(net_disc).requires_grad_(False)`
and
`accelerator.unwrap_model(net_disc).requires_grad_(True)`
should be called at appropriate places.

Thank you for finding this! @onepeachbiubiubiu

**Comment by GaParmar** [comment]
Good catch! Feel free to make a PR changing this, I will merge it!
